### ISSUE TITLE
Adding SHAP-like beeswarm plot

### ISSUE THREAD

**Issue by oasidorshin** [optuna/optuna#4987]
### Motivation

SHAP beeswarm plot looks like this:
![Beeswarm-plot-of-SHAP-calculation-for-the-ten-highest-ranking-variables-Variables-are](https://github.com/optuna/optuna/assets/79442793/c0448959-15bf-4a90-87d3-f832dc1e85ae)

It is very useful for determining monotonic relationships between feature values and model predictions, example in the picture:
higher age -> more willing to get vaccine

### Description

For optuna, similar plot with parameter values on y axis and function value on x axis would also be very useful for exactly the same purpose

### Additional context (optional)

I can work on this issue if the idea is approved

**Comment by contramundum53** [comment]
Thank you for your suggestion!
How about [Parallel Coordinate Plot](https://optuna--5053.org.readthedocs.build/en/5053/reference/visualization/generated/optuna.visualization.plot_parallel_coordinate.html#optuna.visualization.plot_parallel_coordinate)? Do you think beeswarm plot looks better?

**Comment by contramundum53** [comment]
By the way, I like how it avoids putting many points in the same location. Do you think it is easy enough to do with plotly?

**Comment by pangjac** [comment]
> By the way, I like how it avoids putting many points in the same location. Do you think it is easy enough to do with plotly?

When applying parallel coordinate plot and beeswarm plot specifically to plot feature importance, *parallel coordinate plot*, where each line represents a dimension and the lines to show their relationships across dimensions, is a way to show how model features clustered emerge among different features; while in a *beeswarm plot*, the dots are arranged in a way that showing the density and variation of the data specifically for one feature. 

That's my two cents of applying them for different purpose of feature importance plot.  

And yes, I like the idea of doing them with plotly, it is more easier to embed plotly object!

**Comment by contramundum53** [comment]
Sorry I think I'm still not fully understanding your intended plot.

> parameter values on y axis and function value on x 
Do you mean that the colors represent the parameter values and the positions (left to right) represent the objective value?
Then, I think the shapes of density for all parameters will be the same and the only information of this plot will be the colors of each point. 
Is my understanding correct?

If so, how about [Slice Plot](https://optuna.readthedocs.io/en/stable/reference/visualization/generated/optuna.visualization.plot_slice.html)?
Or, if you prefer colored presentation, maybe [Rank Plot](https://optuna.readthedocs.io/en/stable/reference/visualization/generated/optuna.visualization.plot_rank.html#optuna.visualization.plot_rank) is a good choice?

**Comment by oasidorshin** [comment]
@contramundum53 
Yes, parallel coordinate plot is quite similar for this use case, and I also like using it personally, because it can also trace particular paths for individual trials.

Indeed, I like how shap variant avoids putting many points in the same place and is able to efficiently represent sampling density

Another thing that I find problematic about coordinate plot is coloring - it goes from light blue to dark blue, and light blue lines almost disappear visually (ofc I understand that this is Optuna color scheme)

So, I think that parallel coordinate plot and shap beeswarm plot are quite similar - for shap plot, we lose individual paths, but we get densities and because of coloring monotonic relationships between parameter values and objective function values are easier to trace visually.

Because current color Slice and Parallel coordinate plots are serving similar purpose and working quite good, beeswarm may indeed be not needed

**Comment by not522** [comment]
Thank you for your feature request. This feature would be nice to have, but it is need-discussion to put it in the optuna repository. It is easier to put it in OptunaHub, which we recently released, so I submitted the issue there. https://github.com/optuna/optunahub-registry/issues/116

**Comment by y0z** [comment]
Resolved in https://github.com/optuna/optunahub-registry/pull/357.

Available in https://hub.optuna.org/visualization/plot_beeswarm/.
